File: erlang/erf.erl
-module(erf).
-export([erf/1]).

erf(X) ->
    % constants
    A1 =  0.254829592,
    A2 = -0.284496736,
    A3 =  1.421413741,
    A4 = -1.453152027,
    A5 =  1.061405429,
    P  =  0.3275911,

    % Save the sign of x
    Sign = if
        X >= 0 ->  1;
        true   -> -1
    end,
    A = abs(X),

    % A&S formula 7.1.26
    T = 1.0/(1.0 + P*A),
    Y = 1.0 - (((((A5*T + A4)*T) + A3)*T + A2)*T + A1)*T*math:exp(-A*A),

    Sign*Y.

File: pascal/erf.pas
unit erf;

interface

uses math;

function erf(x: float): float;

implementation

function erf(x: float): float;

{ constants }
const
    a1 =  0.254829592;
    a2 = -0.284496736;
    a3 =  1.421413741;
    a4 = -1.453152027;
    a5 =  1.061405429;
    p  =  0.3275911;

var sign: float;
    t, y: float;

begin
    { Save the sign of x }
    sign := 1;
    if x < 0 then
        sign := -1;
    x := abs(x);

    { A&S formula 7.1.26 }
    t := 1.0/(1.0 + p*x);
    y := 1.0 - (((((a5*t + a4)*t) + a3)*t + a2)*t + a1)*t*exp(-x*x);

    erf := sign*y;
end;

end.
